x-pack/plugin/searchable-snapshots/src/internalClusterTest/java/org/elasticsearch/xpack/searchablesnapshots/SearchableSnapshotsIntegTests.java

```diff
@@ -1155,6 +1164,78 @@ private void assertSearchableSnapshotStats(String indexName, boolean cacheEnable
+        createRepository(repositoryName, "fs");
+
+        final SnapshotId snapshotOne = createSnapshot(repositoryName, "snapshot-1", List.of(indexName)).snapshotId();
+        assertAcked(indicesAdmin().prepareDelete(indexName));
+
+        // block the cluster state update thread, so we have the opportunity to inspect the mount/restore tasks' metadata
+        ClusterService clusterService = internalCluster().getCurrentMasterNodeInstance(ClusterService.class);
+        clusterService.submitUnbatchedStateUpdateTask("block master service", new ClusterStateUpdateTask() {
+            @Override
+            public ClusterState execute(ClusterState currentState) {
+                safeAwait(cyclicBarrier);   // notify test that we're blocked
+                safeAwait(cyclicBarrier);   // wait to be unblocked by test
+                return currentState;
+            }
+
+            @Override
+            public void onFailure(Exception e) {
+                throw new AssertionError("block master service", e);
+            }
+        });
+
+        // wait for master thread to be blocked
+        safeAwait(cyclicBarrier);
+
+        ActionFuture<RestoreSnapshotResponse> response = null;
+        try {
+            final MountSearchableSnapshotRequest mountRequest = new MountSearchableSnapshotRequest(
+                indexName,
+                repositoryName,
+                snapshotOne.getName(),
+                indexName,
+                Settings.EMPTY,
+                Strings.EMPTY_ARRAY,
+                true,
+                MountSearchableSnapshotRequest.Storage.FULL_COPY
+            );
+
+            response = client().execute(MountSearchableSnapshotAction.INSTANCE, mountRequest);
+            TaskInfo[] restoreSnapshotTask = new TaskInfo[1];
+            waitUntil(() -> {
+                restoreSnapshotTask[0] = getTaskForActionFromMaster(TransportRestoreSnapshotAction.TYPE.name());
+                return restoreSnapshotTask[0] != null;
+            });
+            TaskInfo mountSnapshotTask = getTaskForActionFromMaster(MountSearchableSnapshotAction.NAME);
+            assertNotNull("No mount task found", mountSnapshotTask);
+            assertEquals(mountSnapshotTask.taskId(), restoreSnapshotTask[0].parentTaskId());
+        } finally {
+            safeAwait(cyclicBarrier);   // Unblock the master thread
+            if (response != null) {
+                response.actionGet();
+                assertAcked(indicesAdmin().prepareDelete(indexName));
+            }
+        }
+    }
+
+    @Nullable
+    private TaskInfo getTaskForActionFromMaster(String action) {
+        var ltr = new ListTasksRequest().setDetailed(true).setNodes(internalCluster().getMasterName()).setActions(action);
+        ListTasksResponse response = client().execute(TransportListTasksAction.TYPE, ltr).actionGet();
+        if (response.getTasks().isEmpty()) {
+            return null;
+        }
+        return response.getTasks().get(0);
```

Comment:
If we run this whole thing in an `assertBusy` then we can just assert that there's exactly one task in the response to trigger the needed retries.